gems/aws-sdk-core/lib/aws-sdk-core/log/param_filter.rb

```diff
@@ -11,33 +11,47 @@ class ParamFilter
       # end
 
       def initialize(options = {})
-        @filters = Set.new(SENSITIVE + Array(options[:filter]))
+        filters = options[:filter] || {}
+        # Support backwards compatibility for Array
+        if filters.is_a?(Array)
+          @filters = SENSITIVE.dup
+          @filters.update(@filters) { |_k, v| v + Array(filters) }
+        elsif filters.is_a?(Hash)
+          @filters = SENSITIVE.merge(filters)
```

Comment:
I think this is an issue that it will override existing params for that service.  Possibly this is useful.... and since the hash is new its not necessarily breaking behavior.  But I think its inconsistent with the way it works for filter arrays (ie. additive).  In either case I think the documentation could use an update to clarify the behavior.

Reply:
We also want a way to disable .. either by overriding or another option. Previously the array was monkey patched in one of the issues.

Reply:
I don't think I can get an option to work because of when the object is created so I think nothing can be additive.